### PR TITLE
[ci-jobs] Add support for github event and include HPOS disabled e2e tests in ci

### LINKED ISSUE #46924: Move HPOS disabled e2e tests into ci.yml
Depends on https://github.com/woocommerce/woocommerce/issues/46925.

The e2e tests with HPOS disabled still run in a separate workflow, with every merge to trunk. We need to move them in `ci.yml` to run as the other tests.

### LINKED ISSUE #46925: Add support to Github events in ci-jobs
Every project in `ci-jobs` runs for any Github event (pull_request, push, etc.) We need to support jobs that only run for certain events.

### PR DESCRIPTION
**Pull request by adimoldovan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/46925
Closes https://github.com/woocommerce/woocommerce/issues/46924

Adds support for Github events in ci-jobs, so we can only run jobs for specific events, like only for `push`.

- Added a new option in job configuration: `events` as an array of strings, which should contain a list of events for which the job should run. If no events list is defined or the list is empty the job will run for any event.
- Added a new argument for the `ci-jobs` cli: `--event` or `-e`. If no argument is passed the tool will consider all jobs and do not filter by event.
- Moved the HPOS disabled e2e tests into ci.yml by configuring a new project with `events: [push]` configuration. This should only run for `push` events and not run for `pull_request`
- Deprecated 2 workflows for HPOS disabled e2e tests: one for push to trunk and one for daily runs (did not even know this existed until now).
- Improved the HPOS disabled env setup. Because you cannot set environment variables in the job's run command I had to update the `env:test:non-hpos` command and the test env setup script to disable the HPOS feature. Also added a print for the HPOS value in the global setup to easily spot if the expected value is set.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run pnpm utils ci-jobs with different arguments and check the output.
1. `pnpm utils ci-jobs --event push` should return all configured jobs, including the HPOS disabled ones (5 e2e and 1 API)
2. `pnpm utils ci-jobs --event pull_request` should return all configured jobs without the 5 shards of "Core e2e tests - HPOS disabled" and without the `Core API tests - HPOS disabled`
3. `pnpm utils ci-jobs` should return all configured jobs including the HPOS disabled ones.

- Check CI, the [Build Matrix step](https://github.com/woocommerce/woocommerce/actions/runs/8834156401/job/24255350060?pr=46922#step:4:22) should call `pnpm utils ci-jobs` with the right event argument.
- Tested in CI that HPOS tests having the `pull_request` event configured were triggered: https://github.com/woocommerce/woocommerce/actions/runs/8838048787?pr=46922
- Tested with e2e set to `push` event only (default value) and they were not triggered: https://github.com/woocommerce/woocommerce/actions/runs/8838873869/job/24271155448?pr=46922

### CODE REVIEW

File: tools/monorepo-utils/src/ci-jobs/index.ts [hunk 1]
```diff
@@ -23,11 +23,24 @@ const program = new Command( 'ci-jobs' )
 		'Base ref to compare the current ref against for change detection. If not specified, all projects will be considered changed.',
 		''
 	)
+	.option(
+		'-e --event <event>',
+		'Github event for which to run the jobs. If not specified, all projects will with changes be considered.',
```
Comment (rrennick) [inline tools/monorepo-utils/src/ci-jobs/index.ts]: The wording here is unclear. Do you mean `If not specified, all events will be considered.`? The current code already considers all packages with changes.

**Review by rrennick — COMMENTED** [review]
> pnpm utils ci-jobs should return all configured jobs without the 5 shards of "Core e2e tests - HPOS disabled"

The response includes the 5 HPOS disabled tests for me. Also, this PR only includes `@woocommerce/plugin-woocommerce - Core API tests - HPOS disabled` when the other HPOS disabled tests are included. The PR testing instructions should be updated to reflect that.

I'm leaving this as a comment to not hang the PR on my review as I'll be AFK next week.

File: tools/monorepo-utils/src/ci-jobs/index.ts (hunk 1 above)
Comment (adimoldovan) [reply tools/monorepo-utils/src/ci-jobs/index.ts]: I was struggling to get the right wording here. Your suggestion is much better. Updated with [5008e80](https://github.com/woocommerce/woocommerce/pull/46922/commits/5008e805cbcaf5a8db8d99fa71496459f5eb8b5a)

**Comment by adimoldovan** [comment]
> > pnpm utils ci-jobs should return all configured jobs without the 5 shards of "Core e2e tests - HPOS disabled"

> The response includes the 5 HPOS disabled tests for me. Also, this PR only includes `@woocommerce/plugin-woocommerce - Core API tests - HPOS disabled` when the other HPOS disabled tests are included. The PR testing instructions should be updated to reflect that.

I fixed the testing instructions. That command should return the HPOS disabled tests, it worked  as expected.

**Review by lanej0 — APPROVED** [review]
LGTM

**Pull request merged**